src/parameter.cpp

```diff
@@ -21,65 +21,63 @@
 using namespace std;
 
 #define ERRORLOG cerr << "ERROR in Parameter: "
-#define TRACELOG if (_verbose) cout << "TRACE in Parameter: "
+#define TRACELOG \
+  if (_verbose)  \
+  cout << "TRACE in Parameter: "
 
-namespace acap_runtime {
-
-// Initialize Parameter Service
-bool Parameter::Init(const bool verbose) {
-  _verbose = verbose;
-  TRACELOG << "Init" << endl;
-  return true;
-}
-
-// Logic and data behind the server's behavior.
-Status Parameter::GetValues(ServerContext* context,
-    ServerReaderWriter<Response, Request>* stream) {
+namespace acap_runtime
+{
+  Parameter::~Parameter()
+  {
+    if (ax_parameter != NULL)
+      ax_parameter_free(ax_parameter);
+  }
 
-    char parhand_result[BUFSIZ];
+  // Initialize Parameter Service
+  bool Parameter::Init(const bool verbose)
+  {
+    _verbose = verbose;
+    TRACELOG << "Init" << endl;
+    _error = NULL;
+    if (ax_parameter == NULL)
+      ax_parameter = ax_parameter_new(APP_NAME, &_error);
+    if (ax_parameter == NULL)
+    {
+      ERRORLOG << "Error when creating axparameter: " << _error->message << endl;
+      g_clear_error(&_error);
+      return false;
+    }
 
-    Request request;
-    while (stream->Read(&request)) {
-      size_t pos = 0;
-      const char *parameter_value= NULL;
-      string parhand_cmd = "parhandclient get ";
-      string parameter_key = request.key().c_str();
-      const regex pattern("[a-zA-Z0-9.]+");
-      if(regex_match(parameter_key, pattern))
-      {
-        string parhandclient_cmd = parhand_cmd + parameter_key;
+    return true;
+  }
 
-        FILE *fp = popen(parhandclient_cmd.c_str(), "r"); 
-        if (!fp){
-          throw std::runtime_error("popen() failed!");
-        }
-        std::string value;
-        if ( fgets( parhand_result, BUFSIZ, fp ) != NULL ) {
-          value = parhand_result;
-          while ((pos = value.find('"', pos)) != std::string::npos)
-            value = value.erase(pos, 1);
-          parameter_value = value.c_str();
-        }
-        if (parameter_value != nullptr){
-          TRACELOG << request.key().c_str() << ": " << parameter_value << endl;
-        }
-        else {
-          parameter_value = "";
-          TRACELOG << request.key().c_str() << ": " << parameter_value << endl;
-        }
+  Status Parameter::GetValues(ServerContext *context, const Request *request, Response *response)
+  {
+    if (ax_parameter == NULL)
```

Comment:
Maybe it actually _is_ worth the effort, because by initializing ax_parameter in the constructor, it is guaranteed to be always initialized.

Reply:
I'm not sure it is - this API has so far followed the same 'pattern' as the two other (i.e. using Init instead of a constructor). I don't see any real benefit in refactoring this API only.